workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSBehaviourLibrary.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Generic;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    /// <summary>
+    /// Retrieves Reader and Writer fields from MonoBehaviours and handles injection into them.
+    /// </summary>
+    public class SpatialOSBehaviourLibrary {
```

Comment:
This seems like it mixes worker wide stuff with gameobject specific stuff. What do we expect to own instances of this?

Reply:
This was initially a static class made non-static for testability. It's probably going to be created by a dispatcher ECS system that calls into the GameObjectRepresentation ECS component's methods, so the same instance will be shared between all gameobjects.

Reply:
How does it do the entity specific stuff then? Like the inject functions.

Reply:
Discussed offline - as far as I understand this will be created in an ECS system, passed to each GameObjectRepresentation ECS component (one per entity), and that component will do the bookkeeping of what can be activated when, calling into the library to 1) know what components it's looking for, 2) do the injections when everything is ready. (Then it enables the monobehaviour.)